test(e2e,playwright): Header

gerhynes opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Addresses #51705 

<!-- Feel free to add any additional description of changes below this line -->
Thanks as always for all the work you put into these educational resources. And for giving me an opportunity to learn Playwright.

Oh, and camperbot cant code

gerhynes committed:
Add Playwright tests for Header component

gerhynes committed:
Reference env vars

huyenltnguyen reviewed (commented):
Hey @gerhynes, thank you for the PR!

We have merged #51835, which initialized and added some tests for the Header. However, your PR has tests that we would like to include as well. 

So please help:
- Update this branch with the latest `main` branch
- Remove the duplicated tests
- Address the PR feedback

And we will get back to the PR as soon as we can 🙂

gerhynes committed:
Reconcile tests

gerhynes commented on e2e/header.spec.ts:
```diff
@@ -0,0 +1,149 @@
+    await logoLink.click();
+    await page.waitForURL(`/learn`);
+    await expect(page).toHaveURL(`/learn`);
+  });
```
I've removed this test since "Renders universal nav by default" already checks for the link to `/learn`.

gerhynes commented on e2e/header.spec.ts:
```diff
@@ -0,0 +1,149 @@
+    for (const menuLink of menuLinks) {
+      const link = page
+        .getByRole('link', { name: menuLink.name })
+        .filter({ hasText: menuLink.name }); // need to distinguish from freeCodeCamp logo link in header
```
Using `exact` works just fine, so I've removed the filter.

gerhynes commented:
Hi @huyenltnguyen, Thanks so much for the feedback. Sorry that it has taken a while to update the PR.

I've removed the duplicate tests, incorporated the PR feedback, and made some changes to reconcile the new and existing tests. For example,
- `test.beforeEach` now directs to `/` since only one of the tests is specific to the C# exam page.
- I removed `test.afterEach` since Playwright already takes care of closing the page.
- I updated the test descriptions to be more consistent.

Thanks for taking the time to look at this PR.

huyenltnguyen reviewed (approved):
LGTM 👍 

@gerhynes Thanks a lot for the improvements and the additional tests. The changes breakdown helped the review process as well 🙂

gerhynes commented:
Thanks so much @huyenltnguyen. Glad I could help.
